### PR TITLE
Rewrite parser

### PR DESCRIPTION
**Pull request by williamtu**
```
 1: datapath - basic BPF commands                   ok
  2: datapath - ping between two ports               ok
  3: datapath - http between two ports               ok
  4: datapath - ping between two ports on vlan       ok
  5: datapath - ping between two ports on cvlan      ok
  6: datapath - ping6 between two ports              ok
  7: datapath - ping6 between two ports on vlan      ok
  8: datapath - ping6 between two ports on cvlan     ok
  9: datapath - ping over bond                       skipped (system-bpf-traffic.at:210)
 10: datapath - ping over vxlan tunnel               ok
 11: datapath - ping over vxlan6 tunnel              ok
 12: datapath - ping over gre tunnel                 ok
 13: datapath - ping over geneve tunnel              ok
 14: datapath - ping over geneve tunnel with TLV     ok
 15: datapath - ping over geneve6 tunnel             ok
 16: datapath - clone action                         FAILED (system-bpf-traffic.at:487)
 17: datapath - mpls actions                         FAILED (system-bpf-traffic.at:526)
 18: datapath - basic truncate action                FAILED (system-bpf-traffic.at:586)
 19: datapath - truncate and output to gre tunnel    FAILED (system-bpf-traffic.at:720)
 20: ovn -- 1 LR connects 2 LSes                     ok
```

### CODE REVIEW

File: bpf/parser.h [hunk 1]
```diff
@@ -16,392 +16,322 @@
  * 02110-1301, USA
  */
 
-/*
- * Protocol parser generated from P4 1.0
- */
 #include "ovs-p4.h"
```
Comment (YiHungWei) [inline bpf/parser.h:19]: Do we still need this header file after we get rid of P4 parser?

File: bpf/parser.h [hunk 2]
```diff
@@ -16,392 +16,322 @@
...
-            goto end;
-        }
-        else
-            goto ovs_tbl_4;
-    }
-    parse_arp: {
-        struct arp_rarp_t *arp = &ebpf_headers.arp;
-        if (skb_load_bytes(skb, offset, arp, sizeof ebpf_headers.arp) < 0) {
-            ebpf_error = p4_pe_header_too_short;
+    if (eth->h_proto == bpf_htons(ETH_P_8021Q)){
```
Comment (yifsun) [inline bpf/parser.h:97]: use eth_proto

File: bpf/parser.h (hunk 1 above)
Comment (williamtu) [reply bpf/parser.h:19]: good catch... I will remove it.

File: bpf/parser.h (hunk 2 above)
Comment (williamtu) [reply bpf/parser.h:97]: ok thanks!

File: bpf/parser.h [hunk 3]
```diff
@@ -16,392 +16,322 @@
...
         goto end;
     }
-    ebpf_headers.valid |= ETHER_VALID;
-    offset += 14;
-    skbOffsetInBits = offset * 8;
+    offset += sizeof(hdrs.ethernet);
+    hdrs.valid |= ETHER_VALID;
 
-    /* vlan_tci is in host byte order. */
+    /* VLAN 8021Q (0x8100) or 8021AD (0x8a88) in metadata
```
Comment (YiHungWei) [inline bpf/parser.h:84]: 8021AD (0x88a8)

Comment (williamtu) [reply bpf/parser.h:84]: ok, thanks!

**Comment by williamtu** [comment]
I will send v2

**Pull request closed without merging**